lookout/style/format/tests/test_analyzer_integration.py

```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+import queue
+import re
+import shutil
+import signal
+import subprocess
+import sys
+import tarfile
+import tempfile
+from threading import Thread
+import unittest
+
+from lookout.core.tests.server import find_port, run
+
+
+class AnalyzerIntegrationTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        base_path = Path(__file__).parent.resolve()
+        cls.base_dir = str(base_path)
+        print(cls.base_dir)
+        print(cls.base_dir)
+        cls.jquery_dir = str(base_path / "jquery")
+        # str() is needed for Python 3.5
+        with tarfile.open(str(base_path / "jquery.tar.xz")) as tar:
+            tar.extractall(path=str(cls.base_dir))
+
+    @classmethod
+    def tearDownClass(cls):
+        shutil.rmtree(cls.jquery_dir)
+
+    def _wait_ready(self):
+        for line in iter(self.analyzer.stdout.readline, b""):
+            line = line.decode("utf-8", "replace")
+            if line.find(":run:Listening localhost:%d" % self.port) != -1:
+                return True
+
+    def _tee(self):
+        for line in iter(self.analyzer.stdout.readline, b""):
+            line = line.decode("utf-8", "replace")
+            line = line.rstrip()
+            print(line, file=sys.stderr)
+            self.analyzer_output.put(line)
+
+    def setUp(self):
+        self.port = find_port()
+        self.db = tempfile.NamedTemporaryFile(dir=self.base_dir)
+        command = "python3 -m lookout init --db sqlite:///%s --fs %s" % (self.db.name,
+                                                                         self.base_dir)
+        subprocess.check_call(command.split(" "))
+        command = ("python3 -m lookout run lookout.style.format --db sqlite:///%s "
+                   "--server localhost:%d --fs %s --log-level DEBUG"
+                   ) % (self.db.name, self.port, self.base_dir)
+        self.analyzer = subprocess.Popen(
+            command.split(" "), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        waiter = Thread(target=self._wait_ready)
+        waiter.start()
+        waiter.join(5)
```

Comment:
We can call `main()` directly without the need for `subprocess`

Reply:
Regarding this comment, do you have something in mind specifically? To me subprocess is the easiest to do the 3 required things: wait for initialization, monitor output, send sigint to cleanup.

Reply:
Yep, calling the main() function directly. Don't get the question... main() is defined in __main__.py

Reply:
I see. It should be possible to find the class instance either through the chain of function locals or the global list of objects. You will be able to shutdown cleanly.